[feature] delayed_sidekiq strategy (#869)
toptal/chewy: README.md

Chewy.settings[:sidekiq] = {queue: :low}
```

#### `:delayed_sidekiq`

> It accumulate ids of records to be reindexed during the latency window in redis and then does the reindex of all accumulated records at once.

Reviewer: ```suggestion
It accumulates ids of records to be reindexed during the latency window in redis and then does the reindexing of all accumulated records at once.
```

The strategy is very useful in case of frequently mutated records.

> It supports update_fields option, so it will try to select just enough data from the DB

Reviewer: ```suggestion
It supports `update_fields` option, so it will try to select just enough data from the DB
```

There are three options that can be defined in the index:
```ruby
class CitiesIndex...
  delayed_sidekiq_options latency: 3, margin: 2, reindex_wrapper: ->(&reindex) { ActiveRecord::Base.connected_to(role: :reading) { reindex.call } }

  # latency - will prevent scheduling identical jobs
  # margin - main purpose is to cover db replication lag by the margin
  # reindex_wrapper - lambda that accepts block to wrap that reindex process AR connection block.
  ...
end
```

```ruby
Chewy.strategy(:delayed_sidekiq) do
  City.popular.map(&:do_some_update_action!)
end
```

The default queue name is `chewy`, you can customize it in settings: `sidekiq.queue_name`
```
Chewy.settings[:sidekiq] = {queue: :low}
```

> Explicit call of the reindex using :delayed_sidekiq strategy

Reviewer: ```suggestion
Explicit call of the reindex using `:delayed_sidekiq strategy`
```

```ruby
CitiesIndex.import([1, 2, 3], strategy: :delayed_sidekiq)
```

> Explicit call of the reindex using :delayed_sidekiq strategy with :update_fields support

Reviewer: ```suggestion
Explicit call of the reindex using `:delayed_sidekiq` strategy with `:update_fields` support
```

Chewy.settings[:sidekiq] = {queue: :low}
```

#### `:delayed_sidekiq`

It accumulates ids of records to be reindexed during the latency window in redis and then does the reindexing of all accumulated records at once.
The strategy is very useful in case of frequently mutated records.
It supports `update_fields` option, so it will try to select just enough data from the DB

There are three options that can be defined in the index:
```ruby
class CitiesIndex...
  strategy_config delayed_sidekiq: {
    latency: 3,
    margin: 2,
    ttl: 60 * 24,
    reindex_wrapper: ->(&reindex) {
      ActiveRecord::Base.connected_to(role: :reading) { reindex.call }
    }
    # latency - will prevent scheduling identical jobs
    # margin - main purpose is to cover db replication lag by the margin

>     # ttl - a chunk expiration time (in minutes)

Reviewer 2: sorry, this one as well please :)

Author: Oh, damn. Thank you! Missed that one.
Now it should be fine.
